resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -93,7 +93,9 @@ object ExecutorPodsSnapshot extends Logging {
       (
         pod.getStatus == null ||
         pod.getStatus.getPhase == null ||
-        pod.getStatus.getPhase.toLowerCase(Locale.ROOT) != "terminating"
+          (pod.getStatus.getPhase.toLowerCase(Locale.ROOT) != "terminating" &&
+           pod.getStatus.getPhase.toLowerCase(Locale.ROOT) != "running" &&
+           pod.getStatus.getPhase.toLowerCase(Locale.ROOT) != "pending")
```

Comment:
Is this exhaustive for that cases?

Reply:
No, but these are the cases which if it's deleted timestamp it might not be deleted. For example if it's in a failed state it's ok if its detected as deleted.